demo/launch.sh

```diff
@@ -0,0 +1,134 @@
+  echo "Starting pool node${node} on port ${port} ..."
+  $CARDANO_NODE run \
+    --config "$TESTNET_DIR/configuration.yaml" \
+    --topology "$TESTNET_DIR/node-data/node${node}/topology.json" \
+    --database-path "$TESTNET_DIR/node-data/node${node}/db" \
+    --shelley-kes-key "$TESTNET_DIR/pools-keys/pool${node}/kes.skey" \
+    --shelley-vrf-key "$TESTNET_DIR/pools-keys/pool${node}/vrf.skey" \
+    --shelley-operational-certificate "$TESTNET_DIR/pools-keys/pool${node}/opcert.cert" \
+    --byron-delegation-certificate "$TESTNET_DIR/pools-keys/pool${node}/byron-delegation.cert" \
+    --byron-signing-key "$TESTNET_DIR/pools-keys/pool${node}/byron-delegate.key" \
+    --socket-path "$TESTNET_DIR/node-data/node${node}/sock" \
+    --host-addr 127.0.0.1 \
+    --port "$port" \
+    &> "$TESTNET_DIR/logs/node${node}/out.log"
+}
+
+# Start the toxiproxy server
+start_toxiproxy_server() {
+  echo "Starting toxiproxy server ..."
+  toxiproxy-server &> "$TESTNET_DIR/logs/toxiproxy.log"
+}
+
+# Create a proxy port for a given node with some latency
+create_toxiproxy() {
+  local node="$1"
+  local in_port="$2"
+  local out_port="$3"
+  local latency="$4"
+  echo "Creating toxiproxy ${in_port} -> ${out_port} ..."
+  toxiproxy-cli create \
+    --listen "127.0.0.1:${out_port}" \
+    --upstream "127.0.0.1:${in_port}" \
+    "node${node}_proxy"
+  toxiproxy-cli toxic add \
+    --type latency \
+    --attribute latency="$latency" \
+    "node${node}_proxy"
+
+}
+
+# "Gracefully" stop all pool nodes on script termination
+cleanup() {
+  echo ""
+  echo "Stopping pool nodes ..."
+  pkill -f "$CARDANO_NODE" || true
+  echo "Stopping toxiproxy server ..."
+  pkill -f toxiproxy-server || true
+  echo "Removing testnet directory $TESTNET_DIR ..."
+  rm -rf "$TESTNET_DIR"
+}
+
+# ---------------------------------
+
+# Export variables needed to enable Peras+CertConjuring in cardano-node
+export ENABLE_PERAS=1
+export PERAS_CERT_BOOST=2
+export PERAS_CERT_CONJURING=1
+export PERAS_CERT_CONJURING_STATIC_FILES="$STATIC_FILES"
+
+# Recreate the testnet environment
+echo "Setting up testnet environment in $TESTNET_DIR ..."
+rm -rf "$TESTNET_DIR"
+mkdir -p "$TESTNET_DIR" "$TESTNET_DIR/logs"
+cp --no-preserve=all -r "$TESTNET_ENV"/* "$TESTNET_DIR"
+chmod 600 "$TESTNET_DIR/pools-keys"/*/*.skey
+
+# Update the "startup timestamp" field in a couple of places
+ISO_8601_TS=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+jq --arg ts "$ISO_8601_TS" \
+  '. + { "systemStart": $ts }' \
+  "$TESTNET_ENV/shelley-genesis.json" > \
+  "$TESTNET_DIR/shelley-genesis.json"
+
+EPOCH_TS=$(date -u +"%s")
+jq --argjson ts "$EPOCH_TS" \
+  '. + { "startTime": $ts }' \
+  "$TESTNET_ENV/byron-genesis.json" > \
+  "$TESTNET_DIR/byron-genesis.json"
+
+# Launch some auxiliary processes
+wait_for_http_server 1 &
+start_toxiproxy_server &
+
+# Launch 3 pool nodes behind toxiproxy instances with some latency to induce
+# network partitions to create forks
+NODE_LATENCY_MS=500
```

Comment:
Minor: This amount of latency (2-3 slots, compared to a mean block time of 20 slots) is still well captured by the $\Delta$ parameter. The term "network partition" is used when blocks between disjoint subsets of nodes take significantly longer than $\Delta$ to diffuse.